controllers/etcdpeer_controller.go

```diff
@@ -331,6 +338,57 @@ func (r *EtcdPeerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
+	if !peer.ObjectMeta.DeletionTimestamp.IsZero() {
+		if hasPvcDeletionFinalizer(peer) {
+			log.V(2).Info("Deleting PVC for peer prior to deletion")
+			expectedPvc := pvcForPeer(&peer)
+			expectedPvcNamespacedName, err := client.ObjectKeyFromObject(expectedPvc)
+			if err != nil {
+				log.Error(err, "unable to get ObjectKey frome PVC")
+				return ctrl.Result{}, nil
+			}
+			var actualPvc corev1.PersistentVolumeClaim
+			err = r.Get(ctx, expectedPvcNamespacedName, &actualPvc)
+			switch {
+			case client.IgnoreNotFound(err) != nil:
+				return ctrl.Result{}, fmt.Errorf("failed to get PVC for deleted peer: %w", err)
+			case err == nil:
+				// PVC exists.
+				// Check whether it has already been deleted (probably by us).
+				// It won't actually be deleted until the garbage collector
+				// deletes the Pod which is using it.
+				if actualPvc.ObjectMeta.DeletionTimestamp.IsZero() {
+					log.V(2).Info("Deleting PVC for peer")
+					err := r.Delete(ctx, expectedPvc)
+					if err == nil {
+						log.V(2).Info("Deleted PVC for peer")
+						return ctrl.Result{}, nil
+					} else {
+						return ctrl.Result{}, fmt.Errorf("failed to delete PVC for peer: %w", err)
+					}
```

Comment:
`return` is used above - you can remove the `else`'s here :)

Reply:
Done.